src/base/rss/rss_session.cpp

```diff
@@ -282,36 +282,61 @@ void Session::load()
 
 void Session::loadFolder(const QJsonObject &jsonObj, Folder *folder)
 {
+    bool updated = false;
     foreach (const QString &key, jsonObj.keys()) {
-        QJsonValue val = jsonObj[key];
+        const QJsonValue val {jsonObj[key]};
         if (val.isString()) {
+            // previous format (reduced form) doesn't contain UID
             QString url = val.toString();
             if (url.isEmpty())
                 url = key;
-            addFeedToFolder(url, key, folder);
+            addFeedToFolder(generateUID(), url, key, folder);
+            updated = true;
         }
-        else if (!val.isObject()) {
-            Logger::instance()->addMessage(
-                        QString("Couldn't load RSS Item '%1'. Invalid data format.")
-                        .arg(QString("%1\\%2").arg(folder->path(), key)), Log::WARNING);
-        }
-        else {
-            QJsonObject valObj = val.toObject();
+        else if (val.isObject()) {
+            const QJsonObject valObj {val.toObject()};
             if (valObj.contains("url")) {
                 if (!valObj["url"].isString()) {
-                    Logger::instance()->addMessage(
-                                QString("Couldn't load RSS Feed '%1'. URL is required.")
-                                .arg(QString("%1\\%2").arg(folder->path(), key)), Log::WARNING);
+                    LogMsg(QString("Couldn't load RSS Feed '%1'. URL is required.")
+                           .arg(QString("%1\\%2").arg(folder->path(), key)), Log::WARNING);
```

Comment:
maybe not related to this PR and is minor, why `\\` is chosen as separator? would using OS native separator a better choice?

Reply:
>why `\\` is chosen as separator? would using OS native separator a better choice?

As I recall, it came from somewhere in the old code. If I change it, I would prefer to stay platform-independent and just use `/`.

>maybe not related to this PR 

Yes. Not here. Maybe later.